mlflow/utils/requirements_utils.py

```diff
@@ -240,12 +258,24 @@ def _infer_requirements(model_uri, flavor):
         "mlflow",
     ]
     packages = _prune_packages(packages) - set(excluded_packages)
-    return ["{}=={}".format(p, _get_package_version_from_metadata(p)) for p in sorted(packages)]
+    return sorted(map(_get_pinned_requirement, packages))
+
+
+def _get_local_version_label(version):
+    """
+    Extracts a local version label from `version`.
+
+    :param version: A version string.
+    """
+    try:
+        return Version(version).local
+    except InvalidVersion:
+        return None
 
 
-def _strip_local_version_identifier(version):
+def _strip_local_version_label(version):
```

Comment:
Renamed `_strip_local_version_identifier` to `_strip_local_version_label`. The following code describes why:

```
1.2.3+abc
     ^^^^ local version label (segment starting with '+')
^^^^^^^^^ local version identifier (version identifier containing a local version label)
```